src/chatd.cpp

```diff
@@ -431,18 +431,34 @@ Promise<void> Connection::reconnect(const std::string& url)
                                                     mUrl.port, (mUrl.path).c_str())), "connect");
                 }
             })
-            .fail([this](const promise::Error& err)
+            .fail([wptr, this](const promise::Error& err)
             {
+                if (wptr.deleted())
+                {
+                    CHATD_LOG_DEBUG("DNS resolution failed, but chatd client was deleted. Error: %s", err.what());
+                    return;
+                }
+
                 if (err.type() == ERRTYPE_MEGASDK)
```

Comment:
This will silently hang the process if (err.type() != ERRTYPE_MEGASDK), and mConnectPromise and mLoginPromise will never be resolved or rejected.

Reply:
Fixed!